lib/ex_unit/lib/ex_unit/filters.ex

```diff
@@ -125,25 +136,34 @@ defmodule ExUnit.Filters do
       {:excluded, "due to foo filter"}
 
   """
-  @spec eval(t, t, map, [ExUnit.Test.t()]) :: :ok | {:error, binary}
+  @spec eval(t, t, map, [ExUnit.Test.t()]) ::
+          :ok | {:excluded, String.t()} | {:skipped, String.t()}
   def eval(include, exclude, tags, collection) when is_map(tags) do
-    skip? = not Enum.any?(include, &has_tag(&1, %{skip: true}, collection))
+    excluded = Enum.find_value(exclude, &has_tag(&1, tags, collection))
+    excluded? = not (!excluded)
+    included? = Enum.any?(include, &has_tag(&1, tags, collection))
+
+    skip_tag =
+      case Map.fetch(tags, :skip) do
```

Comment:
This assignment can be written as `skip_tag = %{skip: Map.get(tags, :skip, true)}`.

Reply:
Since `skip_tag` and `skip_included_explicitely?` are used only on the `if` branch, it should be moved inside the `if` clause.

Reply:
thank you. fixed!